src/submission/submission.service.ts

```diff
@@ -104,11 +105,13 @@ export class SubmissionService {
     }
 
     // Monitoring Plan checks
-    const mpCheckSession = await entityManager
-      .createQueryBuilder(CheckSession, 'cs')
-      .where('cs.mon_plan_id = :monPlanId', { monPlanId: evaluationItem.monPlanId })
-      .andWhere('cs.process_cd = :processCode', { processCode: 'MP' })
-      .getOne();
+    const monPlanData = await entityManager.findOneBy(MonitorPlan, { monPlanIdentifier: evaluationItem.monPlanId });
+    const mpCheckSession = monPlanData.chkSessionIdentifier
+      ? await entityManager.findOneBy(CheckSession, {
+        id: monPlanData.chkSessionIdentifier,
+        processCode: 'MP'
+      })
+      : null;
```

Comment:
May want to check if an MP record is found, first:
```suggestion
    const monPlanData = await entityManager.findOneBy(MonitorPlan, { monPlanIdentifier: evaluationItem.monPlanId });
    if (!monPlanData) {
      throw new EaseyException(new Error(`Monitor Plan record not found for ID: ${evaluationItem.monPlanId}`), HttpStatus.NOT_FOUND);
    }
    const mpCheckSession = monPlanData.chkSessionIdentifier
      ? await entityManager.findOneBy(CheckSession, {
        id: monPlanData.chkSessionIdentifier,
        processCode: 'MP'
      })
      : null;
```